client/controller/index.web.js

```diff
@@ -123,6 +142,11 @@ export function redirectLoggedOut( context, next ) {
 			loginParameters.locale = login_locale;
 		}
 
+		if ( isEditorPath( context.path ) && siteFragment ) {
+			window.location = `https://public-api.wordpress.com/wpcom/v2/redirect?path=${ context.path }&site=${ siteFragment }&redirect=true`;
```

Comment:
I haven't reviewed the related diff's code, but is it possible that one could exploit this endpoint and provide an external URL to `path`? In other words, is this protected from an open redirect?

Furthermore, regarding the `redirect=true` parameter, do we have a use for when it's `false` / not set? If not, can we just remove it and always maintain the redirect behavior?

Reply:
In the endpoint, I do a  search for the site and only redirect to the post if it is a public one. 
If the site doesn't exist then the existing behaviour is preserved. 

If the post is not public then the existing behaviour is preserved. 

This was done to preserve avoid the open redirect issue. 

> regarding the redirect=true parameter, do we have a use for when it's false / not set? If not, can we just remove it and always maintain the redirect behaviour?

If we don't have the 'redirect=true' the endpoint works as a regular endpoint and just returns the json output.
Which currently isn't used anywhere. 

I think we can always use the redirect behaviour but I wanted the endpoint to work as an endpoint and not just a URL that you point folks to. I am happy to simplify it I just thought I would be consistent by default.